chips/lowrisc/src/gpio.rs

```diff
@@ -0,0 +1,254 @@
+        };
+    }
+
+    /* OpenTitan doesn't appear to support floating state GPIOs */
+    fn set_floating_state(&self, _mode: hil::gpio::FloatingState) {
+        // TODO
+    }
+
+    fn floating_state(&self) -> hil::gpio::FloatingState {
+        // TODO
+        hil::gpio::FloatingState::PullNone
+    }
+
+    fn deactivate_to_low_power(&self) {
+        self.disable_input();
+        self.disable_output();
+    }
+
+    fn make_output(&self) -> hil::gpio::Configuration {
+        let regs = self.registers;
+
+        regs.direct_oe.modify(self.set);
+
+        if self.set.value < 15 {
+            regs.masked_oe_low.set(self.set.value << 16);
+            regs.masked_oe_low.modify(self.set);
+        } else {
+            regs.masked_oe_upper.set((self.set.value - 16) << 16);
+            regs.masked_oe_upper.set(self.set.value - 16);
+        }
+
+        self.configuration()
+    }
+
+    fn disable_output(&self) -> hil::gpio::Configuration {
+        let regs = self.registers;
+
+        regs.direct_oe.modify(self.clear);
+
+        if self.set.value < 15 {
+            regs.masked_oe_low.set(self.set.value << 16);
+            regs.masked_oe_low.modify(self.clear);
+        } else {
+            regs.masked_oe_upper.set((self.set.value - 16) << 16);
+            regs.masked_oe_upper.set(self.clear.value - 16);
+        }
+
+        self.configuration()
+    }
+
+    fn make_input(&self) -> hil::gpio::Configuration {
+        let regs = self.registers;
+
+        regs.direct_oe.modify(self.clear);
+
+        if self.set.value < 15 {
+            regs.masked_oe_low.set(self.set.value << 16);
+            regs.masked_oe_low.modify(self.clear);
+        } else {
+            regs.masked_oe_upper.set((self.set.value - 16) << 16);
+            regs.masked_oe_upper.set(self.clear.value - 16);
+        }
+
+        self.configuration()
+    }
+
+    fn disable_input(&self) -> hil::gpio::Configuration {
+        let regs = self.registers;
+
+        regs.direct_oe.modify(self.clear);
+
+        self.configuration()
+    }
+}
+
+impl hil::gpio::Input for GpioPin {
+    fn read(&self) -> bool {
+        let regs = self.registers;
+
+        regs.data_in.is_set(self.pin)
+    }
+}
+
+impl hil::gpio::Output for GpioPin {
+    fn toggle(&self) -> bool {
+        let regs = self.registers;
+
+        let current_outputs = regs.direct_out.extract();
+        if current_outputs.is_set(self.pin) {
+            regs.direct_out.modify_no_read(current_outputs, self.clear);
+        } else {
+            regs.direct_out.modify_no_read(current_outputs, self.set);
+        }
+        regs.direct_out.extract().is_set(self.pin)
+    }
+
+    fn set(&self) {
+        let regs = self.registers;
+
+        regs.direct_out.modify(self.set);
+    }
+
+    fn clear(&self) {
+        let regs = self.registers;
+
+        regs.direct_out.modify(self.clear);
+    }
+}
+
+/* This actually needs to be done */
+impl hil::gpio::Interrupt for GpioPin {
```

Comment:
Is this the only thing left to implement?

Reply:
We've already implemented GPIO interrupts in our branch.

Reply:
I guess this is the only thing left. If @pfmooney has already done it maybe you want to split out a patch to add support? Otherwise I can look into it when I'm back in the office.

Reply:
We agreed in the recent core call that GPIO interrupts weren't needed to merge. There was confidence that work on opentitan would continue and this isn't a fire-and-forget PR. :)